lib/membrane_rtc_engine/engine.ex

```diff
@@ -1175,8 +1184,9 @@ defmodule Membrane.RTC.Engine do
         endpoint = Map.get(endpoints, endpoint_id)
 
         has_subscription_on_track = fn track_id ->
-          subscribed_tracks = Map.get(state.subscriptions, endpoint_id)
-          subscribed_tracks != nil and Map.has_key?(subscribed_tracks, track_id)
+          state.subscriptions
+          |> Map.fetch!(endpoint_id)
+          |> MapSet.member?(track_d)
```

Comment:
We need to create entry in `state.subscriptions` for `endpoint_id` when it is being added and remove this entry when it is being removed.